### ISSUE TITLE
Pass filename to the replace function?

### ISSUE THREAD

**Issue by yangg** [lazd/gulp-replace#50]
Could you pass an argument filename to the replace function ?

**Comment by lazd** [comment]
This is a very useful enhancement, and something that we should definitely include before 1.0.0 to avoid a breaking change.

This would change the function signature for regex replace, which way introduce an incompatibility with streaming regex replace. We'll have to explore further to know.

**Comment by keradus** [comment]
Using stream as source and destination of replacement is great.
If you decided to add a file, please, could you made it as optional parameter or handled by second, dedicated method? It would be really great (and won't break interface).

**Comment by AStoker** [comment]
So I'm looking into trying and get that file functionality implemented. Since it technically is optional, I'm thinking of adding a RegExp option in the Options object to allow the user to specify with a regex string which files will receive this replace treatment.

Having it in the options object would keep any existing users from receiving breaking changes.

**Comment by AStoker** [comment]
However, if you're in need of this kind of functionality now, this plugin should accomplish what you need. https://github.com/sindresorhus/gulp-filter

**Comment by keradus** [comment]
Yeah, and it keep everything KISS ;)

**Comment by ccbayer** [comment]
+1 for this enhancement...

i am working on something similar whereby i replace a string with the current file name in the pipe. i'm not having a ton of luck, but it seems `gulp-tap` is getting me the closest:

```
gulp.task('replaceWithCurrentUrl', function() {
  gulp.src('app/*.html')
  .pipe($.tap(function(file) {
      var fileName = path.basename(file.path);
      gulp.src(file.path)
      .pipe($.replace('{{currentUrl}}', fileName));
      console.log('File in pipe is ' + file.path); 
      console.log('replacing {{currentUrl}} with ' + fileName);
  }))
  .pipe(gulp.dest('app'));
});
```

**Comment by shawnjacobson** [comment]
I cannot get this to work at all, just trying to replace a file path from the server with a local one for debug...

```
gulp.task('server-debug', function () {
    const f = filter(['*', './src/app/models/hist.js'], {restore: true});

    return gulp.src(['./src/**/*','!./src/ngClient/**/*','!./src/ngClient'])
        .pipe(f)
        .pipe(replace('/var/lib/dbs/', 'C:/dbs/'))
        .pipe(f.restore)
        .pipe(gulp.dest('./deploy'));
});
```

I've tried it with and without the filter, tried wrapping the original text in new String('...') to prevent thinking its a RegEx, even tried a completely different string with no special characters, nothing is working.  Any thoughts???

**Comment by lazd** [comment]
Closed by 99474b37d9a983b11828dc4075bd0a98cffd2491.
